src/dbnode/persist/fs/commitlog/files.go

```diff
@@ -27,20 +27,43 @@ import (
 	"sort"
 	"time"
 
+	"github.com/m3db/m3/src/dbnode/persist"
 	"github.com/m3db/m3/src/dbnode/persist/fs"
 	"github.com/m3db/m3/src/dbnode/persist/fs/msgpack"
 )
 
-// File represents a commit log file and its associated metadata.
-type File struct {
-	FilePath string
-	Start    time.Time
-	Duration time.Duration
-	Index    int64
+// NextFile returns the next commitlog file.
+func NextFile(opts Options) (string, int, error) {
+	files, _, err := Files(opts)
+	if err != nil {
+		return "", -1, err
+	}
+
+	newIndex := 0
+	for _, f := range files {
+		if int(f.Index) >= newIndex {
+			newIndex = int(f.Index + 1)
+		}
+	}
+
+	for ; ; newIndex++ {
+		var (
+			prefix   = opts.FilesystemOptions().FilePathPrefix()
+			filePath = fs.CommitlogFilePath(prefix, time.Unix(0, 0), newIndex)
```

Comment:
Can you make `time.Unix(0,0)` a var at the top of this file and put a comment perhaps next to where you use it about why it's no longer required to be any value and can be zero?

```go
var timeNone = time.Unix(0,0)
```

Reply:
sure, done. In the future we can just remove it entirely which I'm pretty sure is backwards compatible (because we just list all the files in the directory then read their heads and ignore their filenames now) but I didn't want to tackle it in this P.R